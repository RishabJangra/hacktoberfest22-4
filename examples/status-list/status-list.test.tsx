import React from 'react';
import renderer from 'react-test-renderer';
import { StatusListScreen } from './StatusList';
import { InfoListItemProps, InfoListItem, wrapIcon } from '@pxblue/react-native-components';
import { FlatList } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

const NotificationIcon = wrapIcon({ IconClass: MaterialIcons, name: 'notifications' });

jest.mock('@react-navigation/native', () => ({
    useNavigation: (): any => ({ openDrawer: jest.fn(() => true) }),
}));
const navigationMock: any = {
    navigate: (): void => {
        /* do nothing */
    },
};

const hardcodedData: InfoListItemProps[] = [
    {
        title: 'Item 1',
        subtitle: 'Subtitle 1',
        iconColor: 'white',
        statusColor: 'red',
        IconClass: NotificationIcon,
    },
    {
        title: 'Item 2',
        subtitle: 'Subtitle 2',
        iconColor: 'red',
        statusColor: 'transparent',
        IconClass: NotificationIcon,
    },
    {
        title: 'Item 3',
        subtitle: 'Subtitle 3',
        iconColor: 'rgba(255,255,255,0.5)',
        statusColor: '#000000',
    },
    {
        title: 'Item 4',
        subtitle: 'Subtitle 4',
        iconColor: 'purple',
        avatar: false,
    },
    {
        title: 'Item 5',
    },
];

describe('Status List Tests', () => {
    it('Screen Renders', () => {
        const tree = renderer
            .create(<StatusListScreen hardcodedData={hardcodedData} navigation={navigationMock} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render 10 items by default', () => {
        const instance = renderer.create(<StatusListScreen navigation={navigationMock} />).root;
        const flatList = instance.findByType(FlatList);
        const infoListItems = flatList.findAllByType(InfoListItem);
        expect(infoListItems).toHaveLength(10);
    });

    it('Should render 5 items if 5 items are passed in', () => {
        const instance = renderer.create(<StatusListScreen hardcodedData={hardcodedData} navigation={navigationMock} />)
            .root;
        const flatList = instance.findByType(FlatList);
        const infoListItems = flatList.findAllByType(InfoListItem);
        expect(infoListItems).toHaveLength(5);
    });
});
