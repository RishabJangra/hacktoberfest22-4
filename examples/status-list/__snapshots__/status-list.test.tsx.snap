// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status List Tests renders the screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    accessible={true}
    emulateUnlessSupported={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "height": 132,
        "shadowColor": "rgba(0, 0, 0, 0.3)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": "100%",
      }
    }
    testID="header"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 112,
          "paddingBottom": 32,
          "paddingHorizontal": 16,
          "paddingVertical": 32,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 80,
            "margin": -16,
            "marginRight": 24,
            "opacity": 1,
            "padding": 16,
            "width": 80,
          }
        }
        testID="header-navigation"
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "marginRight": 0,
            "marginTop": 0,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "justifyContent": "center",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontFamily": "System",
                "fontSize": 20,
                "lineHeight": 20,
                "textAlign": "auto",
                "writingDirection": "ltr",
              }
            }
            testID="header-title"
          >
            Status List
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 112,
              "position": "absolute",
              "right": 8,
            },
            undefined,
          ]
        }
      />
    </View>
  </RCTSafeAreaView>
  <RCTScrollView
    data={
      Array [
        Object {
          "IconClass": [Function],
          "iconColor": "white",
          "statusColor": "red",
          "subtitle": "Subtitle 1",
          "title": "Item 1",
        },
        Object {
          "IconClass": [Function],
          "iconColor": "red",
          "statusColor": "transparent",
          "subtitle": "Subtitle 2",
          "title": "Item 2",
        },
        Object {
          "iconColor": "rgba(255,255,255,0.5)",
          "statusColor": "#000000",
          "subtitle": "Subtitle 3",
          "title": "Item 3",
        },
        Object {
          "avatar": false,
          "iconColor": "purple",
          "subtitle": "Subtitle 4",
          "title": "Item 4",
        },
        Object {
          "title": "Item 5",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 144,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "red",
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 80,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "red",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  },
                  undefined,
                ]
              }
            >
              <Text />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Item 1
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {},
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                Subtitle 1
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.12)",
                    "height": 0.5,
                  },
                  Object {
                    "marginLeft": 72,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 144,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 80,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  },
                  undefined,
                ]
              }
            >
              <Text />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Item 2
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {},
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                Subtitle 2
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.12)",
                    "height": 0.5,
                  },
                  Object {
                    "marginLeft": 72,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 144,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#000000",
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Item 3
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {},
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                Subtitle 3
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.12)",
                    "height": 0.5,
                  },
                  Object {
                    "marginLeft": 72,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 144,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Item 4
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {},
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                Subtitle 4
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.12)",
                    "height": 0.5,
                  },
                  Object {
                    "marginLeft": 72,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 144,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Item 5
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.12)",
                    "height": 0.5,
                  },
                  Object {
                    "marginLeft": 72,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
